pandas/tests/frame/test_sorting.py

```diff
@@ -598,3 +598,86 @@ def test_sort_index_intervalindex(self):
             closed='right')
         result = result.columns.levels[1].categories
         tm.assert_index_equal(result, expected)
+
+    @pytest.mark.parametrize("categories", [['A', 'B', 'C']])
+    @pytest.mark.parametrize("category_indices", [[0, 2, 4]])
+    @pytest.mark.parametrize("list_of_nans", [[np.nan, np.nan]])
+    @pytest.mark.parametrize("na_indices", [[1, 3]])
+    @pytest.mark.parametrize("na_position_first", ['first'])
+    @pytest.mark.parametrize("na_position_last", ['last'])
+    @pytest.mark.parametrize("column_name", ['c'])
```

Comment:
Still not quite it as this will generate the Cartesian product of all of the items here.

Giving it a second look, this might be more complicated than I originally thought with parametrization to the point that it might overcomplicated things. I'd be OK with reverting to what you had before my request, and this could maybe be a separate PR if it even makes sense.

Sorry for the back and forth

Reply:
no problem. I have reverted the parametrization.  Thanks for the careful checks.